application.go

```diff
@@ -732,3 +774,86 @@ func buildURIWithForceParam(path string, force bool) string {
 func buildURI(path string) string {
 	return fmt.Sprintf("%s/%s", marathonAPIApps, trimRootPath(path))
 }
+
+// EmptyLabels explicitly empties labels -- use this if you need to empty
+// labels of an application that already has ip per task with labels defined
+func (i *IPAddressPerTask) EmptyLabels() *IPAddressPerTask {
+
+	i.Labels = &map[string]string{}
+
+	return i
+}
+
+// AddLabel adds a label to an IPAddressPerTask
+//    name: The label name
+//   Value: The label value
+func (i *IPAddressPerTask) AddLabel(name, value string) *IPAddressPerTask {
+
+	if i.Labels == nil {
+		i.EmptyLabels()
+	}
+
+	(*i.Labels)[name] = value
+
+	return i
+}
+
+// EmptyGroups explicitly empties groups -- use this if you need to empty
+// groups of an application that already has ip per task with groups defined
+func (i *IPAddressPerTask) EmptyGroups() *IPAddressPerTask {
+
+	i.Groups = &[]string{}
+
+	return i
+}
+
+// AddGroup adds a group to an IPAddressPerTask
+//  group: The group name
+func (i *IPAddressPerTask) AddGroup(group string) *IPAddressPerTask {
+	if i.Groups == nil {
+		i.EmptyGroups()
+	}
+
+	groups := *i.Groups
+	groups = append(groups, group)
+	i.Groups = &groups
+
+	return i
+}
+
+// EmptyDiscovery explicitly empties discovey data -- use this if you need to empty
+// discovey of an application that already has ip per task with discovey defined
+func (i *IPAddressPerTask) EmptyDiscovery() *IPAddressPerTask {
+	i.Discovery = nil
```

Comment:
This should not be nil but an empty `Discovery` struct.

I'm presuming here that the empty value clears a discovery setting. If it does not, we should drop the method as well.